src/meta/src/manager/catalog/mod.rs

```diff
@@ -127,10 +154,11 @@ where
 
     pub async fn create_database(&self, database: &Database) -> MetaResult<NotificationVersion> {
         let core = &mut self.core.lock().await.database;
-        if !core.has_database(database) {
-            let mut transaction = Transaction::default();
-            database.upsert_in_transaction(&mut transaction)?;
-            let mut schemas = vec![];
+        let mut databases = BTreeMapTransaction::new(&mut core.databases);
+        let mut schemas = BTreeMapTransaction::new(&mut core.schemas);
+        if !databases.contains_key(&database.id) {
```

Comment:
Should check database name duplicate rather than database id, which is always new generated.
```rust
        if core.has_database_key(&database.name) {
            return Err(MetaError::catalog_duplicated("database", &database.name));
        }
        let mut databases = BTreeMapTransaction::new(&mut core.databases);
        let mut schemas = BTreeMapTransaction::new(&mut core.schemas);
        // ... ...
```